src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -870,17 +874,147 @@ private void WriteArrayInfo(XmlBinaryNodeType nodeType, int count)
             WriteMultiByteInt32(count);
         }
 
-        public unsafe void UnsafeWriteArray(XmlBinaryNodeType nodeType, int count, byte* array, byte* arrayMax)
+        public unsafe void UnsafeWriteBoolArray(bool[] array, int offset, int count)
```

Comment:
Nit: All of these UnsafeWriteXxArray methods don't need the type in their names... they could just be UnsafeWriteArray, each of which takes a different type of array

Reply:
I believe the intent was to follow the pattern used by WriteDateTimeArray, WriteGuidArray, and WriteTimeSpanArray, which were already existing in the code.